### PR TITLE
[feat] add dynamic_embedding.embedding_lookup_unique API.

### PR DESCRIPTION
**Pull request by funsimple**
# Description

Brief Description of the PR:

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [x] New Feature

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Recommenders Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/dynamic_embedding_ops_test.py [hunk 1]
```diff
@@ -694,6 +694,30 @@ def _evaluate(tensors, feed_dict):
                           re)
 
 
+@test_util.deprecated_graph_mode_only
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/dynamic_embedding_ops_test.py:697]: plz add UTs in dynamic_embedding_optimizer_test.py for this API.

**Comment by Mr-Nineteen** [comment]
Is there any optimization for Unique?

File: docs/api_docs/dynamic_embedding/embedding_lookup_unique.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+Version of embedding_lookup that avoids duplicate lookups.
+  This can save communication in the case of repeated ids.
+  Same interface as embedding_lookup.
+
+#### Args:
+
+* <b>`params`</b>: A dynamic_embedding.Variable instance.
+* <b>`ids`</b>: a tensor with any shape as same dtype of params.key_dtype.
```
Comment (rhdong) [inline docs/api_docs/dynamic_embedding/embedding_lookup_unique.md:27]: `A tensor `, pls fix the homologous errors in comments

File: tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_ops.py [hunk 3]
```diff
@@ -503,7 +503,58 @@ def initial_value():
...
+                            validate_indices=None,
+                            max_norm=None,
+                            return_trainable=False):
+  """Version of embedding_lookup that avoids duplicate lookups.
+  This can save communication in the case of repeated ids.
+  Same interface as embedding_lookup.
+
+  Args:
+    params: A dynamic_embedding.Variable instance.
+    ids: a tensor with any shape as same dtype of params.key_dtype.
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_ops.py:519]: A tensor

**Comment by rhdong** [comment]
LGTM

**Pull request merged**